railgun/test/integration/controller_test.go

```diff
@@ -0,0 +1,54 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/manager"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHealthEndpoint(t *testing.T) {
+	_ = proxyReady()
+	assert.Eventually(t, func() bool {
+		healthzURL := fmt.Sprintf("http://localhost:%v/healthz", manager.HealthzPort)
+		resp, err := httpc.Get(healthzURL)
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", healthzURL, err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			return true
+		}
+		return false
+	}, ingressWait, waitTick)
+}
+
+func TestMetricsEndpoint(t *testing.T) {
+	if useLegacyKIC() {
+		// The metrics endpoint was intentionally changed for 2.0. Skip if legacy.
+		return
+	}
+	_ = proxyReady()
+	assert.Eventually(t, func() bool {
+		metricsURL := fmt.Sprintf("http://localhost:%v/metrics", manager.MetricsPort)
+		resp, err := httpc.Get(metricsURL)
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", metricsURL, err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			b := new(bytes.Buffer)
+			b.ReadFrom(resp.Body)
+			return strings.Contains(b.String(), "controller_runtime_active_workers")
```

Comment:
Instead of checking that a certain metric is present in the output, we could check whether the output is in the Prometheus format and that it contains a non-zero number of metrics.

This is desirable because:
- before the requested change, the test will break when a metric changes its name, but (likely) not when the protocol changes in a breaking way
- after this change, the test will do the very opposite, which is much more valuable integration feedback

```go
import (
       "github.com/prometheus/common/expfmt"
       "github.com/prometheus/common/model"
)
```

```go
decoder := expfmt.SampleDecoder{
  Dec:  expfmt.NewDecoder(resp.Body, expfmt.FmtText),
  Opts: &expfmt.DecodeOptions{},
}

var v model.Vector
if err := decoder.Decode(&v); err != nil {
  t.Logf("decoder failed: %v", err)
  return false
}

return len(v) > 0
```